File: 1624/solution.java
import java.util.Arrays;

class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        int[] locs = new int[26];
        Arrays.fill(locs, -1);
        int res = -1;
        for (int i = 0; i < s.length(); i++) {
            int c = s.charAt(i) - 'a';
            if (locs[c] != -1) {
                res = Math.max(i - locs[c] - 1, res);
            } else {
                locs[c] = i;
            }
        }
        return res;
    }
}

File: 1624/solution.ts
function maxLengthBetweenEqualCharacters(s: string): number {
  const locs = new Array(26).fill(-1);
  let res = -1;
  for (let i = 0; i < s.length; i++) {
    let c = s[i].charCodeAt(0) - "a".charCodeAt(0);
    if (locs[c] != -1) {
      res = Math.max(i - locs[c] - 1, res);
    } else {
      locs[c] = i;
    }
  }
  return res;
}
